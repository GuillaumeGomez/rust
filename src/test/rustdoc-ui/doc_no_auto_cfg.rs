// edition: 2021
#![deny(warnings)]
#![doc(no_auto_cfg)] //~ ERROR
//~^ ERROR
#![doc(auto_cfg, no_auto_cfg)] //~ ERROR
#![doc(no_auto_cfg(1))] //~ ERROR
#![doc(no_auto_cfg = 1)] //~ ERROR
#![doc(auto_cfg(1))] //~ ERROR
#![doc(auto_cfg = 1)] //~ ERROR

#[doc(auto_cfg)] //~ ERROR
//~^ WARN
#[doc(no_auto_cfg)] //~ ERROR
//~^ WARN
pub struct Bar;
