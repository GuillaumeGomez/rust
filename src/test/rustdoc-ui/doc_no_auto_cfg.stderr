error: this attribute can only be applied at the crate level
  --> $DIR/doc_no_auto_cfg.rs:11:7
   |
LL | #[doc(auto_cfg)]
   |       ^^^^^^^^
   |
note: the lint level is defined here
  --> $DIR/doc_no_auto_cfg.rs:2:9
   |
LL | #![deny(warnings)]
   |         ^^^^^^^^
   = note: `#[deny(invalid_doc_attributes)]` implied by `#[deny(warnings)]`
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #82730 <https://github.com/rust-lang/rust/issues/82730>
   = note: read <https://doc.rust-lang.org/nightly/rustdoc/the-doc-attribute.html#at-the-crate-level> for more information
help: to apply to the crate, use an inner attribute
   |
LL | #![doc(auto_cfg)]
   | ~~~~~~~~~~~~~~~~~

error: this attribute can only be applied at the crate level
  --> $DIR/doc_no_auto_cfg.rs:13:7
   |
LL | #[doc(no_auto_cfg)]
   |       ^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #82730 <https://github.com/rust-lang/rust/issues/82730>
   = note: read <https://doc.rust-lang.org/nightly/rustdoc/the-doc-attribute.html#at-the-crate-level> for more information
help: to apply to the crate, use an inner attribute
   |
LL | #![doc(no_auto_cfg)]
   | ~~~~~~~~~~~~~~~~~~~~

error: `doc(no_auto_cfg)` is enabled by default before the 2024 edition
  --> $DIR/doc_no_auto_cfg.rs:3:8
   |
LL | #![doc(no_auto_cfg)]
   |        ^^^^^^^^^^^
   |
   = note: `#[deny(unused_attributes)]` implied by `#[deny(warnings)]`

error: `doc(no_auto_cfg)` is enabled by default before the 2024 edition
  --> $DIR/doc_no_auto_cfg.rs:5:18
   |
LL | #![doc(auto_cfg, no_auto_cfg)]
   |                  ^^^^^^^^^^^

error: please don't specify both `doc(auto_cfg)` and `doc(no_auto_cfg)`
  --> $DIR/doc_no_auto_cfg.rs:3:8
   |
LL | #![doc(no_auto_cfg)]
   |        ^^^^^^^^^^^
LL |
LL | #![doc(auto_cfg, no_auto_cfg)]
   |                  ----------- the other is specified here

error: `#![doc(no_auto_cfg)]` attribute doesn't expect a value
  --> $DIR/doc_no_auto_cfg.rs:6:8
   |
LL | #![doc(no_auto_cfg(1))]
   |        ^^^^^^^^^^^^^^

error: `#![doc(no_auto_cfg)]` attribute doesn't expect a value
  --> $DIR/doc_no_auto_cfg.rs:7:8
   |
LL | #![doc(no_auto_cfg = 1)]
   |        ^^^^^^^^^^^^^^^

error: `#![doc(auto_cfg)]` attribute doesn't expect a value
  --> $DIR/doc_no_auto_cfg.rs:8:8
   |
LL | #![doc(auto_cfg(1))]
   |        ^^^^^^^^^^^

error: `#![doc(auto_cfg)]` attribute doesn't expect a value
  --> $DIR/doc_no_auto_cfg.rs:9:8
   |
LL | #![doc(auto_cfg = 1)]
   |        ^^^^^^^^^^^^

error: aborting due to 9 previous errors

